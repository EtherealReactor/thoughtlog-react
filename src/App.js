import React, { Component } from 'react';
import { Switch, Route } from 'react-router-dom';
import ProtectedRoute from './ProtectedRoute/ProtectedRoute';
import { connect } from 'react-redux';
import { withRouter } from 'react-router-dom';
import Notification from './Notifications/Notification';
import Welcome from './Welcome/Welcome';
import Thoughts from './Thoughts/Thoughts';
import { closeNotification } from './Notifications/NotificationActions';


class App extends Component {
  
  handleClose = () => {
    console.log('coming to close click');
    this.props.notificationRead();
  }
  
  render() {
    return (
      <React.Fragment>
        <Notification message={this.props.message} handleClose={this.handleClose}/>  
        <Switch>
          <ProtectedRoute path="/thoughts" component={Thoughts} />
          <Route path='/' render={(props) => ( <Welcome {...props} />)} />
        </Switch>
      </React.Fragment>
    );
  }
}

const mapStateToProps = (state) => {
  return {
    message: state.notification.message
  }
}

const mapDispatchToProps = (dispatch) => {
  return {
    notificationRead: () => dispatch(closeNotification())
  }
}

export default withRouter(connect(mapStateToProps, mapDispatchToProps)(App));
// export default connect(mapStateToProps, mapDispatchToProps)(App);
