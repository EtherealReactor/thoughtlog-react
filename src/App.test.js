import React from 'react';
import { shallow } from 'enzyme';
import toJson from 'enzyme-to-json';
import App from './App';


describe("Component === App", () => {
  it("should match snapshot", () => {
    let wrapper = shallow(<App />);
    expect(toJson(wrapper)).toMatchSnapshot();
  });

  it("must contain a div", () => {
    let wrapper = shallow(<App />);
    expect(wrapper.find('div')).toHaveLength(1);
  });

  it("must contain a welcome component", () => {
    let wrapper = shallow(<App />);
    expect(wrapper.find('Welcome')).toHaveLength(1);
  });
});