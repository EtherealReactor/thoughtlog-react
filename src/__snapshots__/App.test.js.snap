// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Component === App should match snapshot 1`] = `
<div>
  <Switch>
    <Route
      path="/thoughts"
      render={[Function]}
    />
    <Route
      path="/"
      render={[Function]}
    />
  </Switch>
</div>
`;
